async_service/function_service/utils.py

```diff
@@ -0,0 +1,83 @@
+import inspect
+import json
+import uuid
+from typing import Any, Callable, Dict
+
+import pydantic
+from pydantic import BaseModel
+
+from async_service.types import Input, InputMessage
+
+INTERNAL_FUNCTION_NAME = "internal_func_name"
+
+
+def create_pydantic_model_from_function_signature(func, model_name: str):
+    # https://github.com/pydantic/pydantic/issues/1391
+    (
+        args,
+        _,
+        varkw,
+        defaults,
+        kwonlyargs,
+        kwonlydefaults,
+        annotations,
+    ) = inspect.getfullargspec(func)
+    defaults = defaults or []
+    args = args or []
+    if len(args) > 0 and args[0] == "self":
+        del args[0]
+
+    non_default_args = len(args) - len(defaults)
+    defaults = [
+        ...,
+    ] * non_default_args + list(defaults)
+
+    keyword_only_params = {
+        param: kwonlydefaults.get(param, Any) for param in kwonlyargs
+    }
+    params = {}
+    for param, default in zip(args, defaults):
+        params[param] = (annotations.get(param, Any), default)
+
+    class Config:
+        extra = "allow"
+
+    # Allow extra params if there is a **kwargs parameter in the function signature
+    config = Config if varkw else None
+
+    return pydantic.create_model(
+        model_name,
+        **params,
+        **keyword_only_params,
+        __base__=pydantic.BaseModel,
+        __config__=config,
+    )
+
+
+def get_functions_dict_with_input_signatures(functions_dict: Dict[str, Callable]):
+    return {
+        name: create_pydantic_model_from_function_signature(func, name).schema()
+        for name, func in functions_dict.items()
+    }
+
+
+async def send_request_to_queue(
+    request_id: str, input: BaseModel, input_publisher: Input
+):
+    my_dict = dict(input._iter(to_dict=False))
+    my_dict[INTERNAL_FUNCTION_NAME] = input.__class__.__name__
+    input_message = InputMessage(request_id=request_id, body=my_dict)
+
+    await input_publisher.publish_input_message(
+        request_id=request_id,
+        serialized_output_message=json.dumps(input_message.dict()).encode(),
```

Comment:
we should probably be explicit here
```suggestion
        serialized_output_message=json.dumps(input_message.dict()).encode("utf-8"),
```